olive/passes/olive_pass.py

```diff
@@ -344,13 +344,17 @@ def run(
 
         # Optimization pass still works on individual graphs.
         if isinstance(model, DistributedOnnxModel):
-            output_filepaths = []
-            for rank in range(model.ranks):
+            for rank in range(model.num_ranks):
                 input_rank_model = model.load_model(rank)
-                rank_output_path = Path(output_model_path).with_suffix("") / str(rank)
-                output_rank_model = self._run_for_config(input_rank_model, data_root, config, rank_output_path)
-                output_filepaths.append(output_rank_model.model_path)
-            output_model = DistributedOnnxModel(output_filepaths, inference_settings=model.inference_settings)
+                rank_output_path = Path(output_model_path).with_suffix("") / model.ranked_model_name(rank)
+                self._run_for_config(input_rank_model, data_root, config, rank_output_path)
+
+            output_model = DistributedOnnxModel(
+                model_path=output_model_path,
```

Comment:
It might be safer to also use `Path(output_model_path).with_suffix("")` here? Just in case the `output_model_path` had a suffix?

Reply:
Done!